pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -145,24 +146,43 @@ def _configure_checkpoint_callbacks(self, checkpoint_callback: bool) -> None:
         if not self._trainer_has_checkpoint_callbacks() and checkpoint_callback is True:
             self.trainer.callbacks.append(ModelCheckpoint())
 
-    def _configure_model_summary_callback(self, weights_summary: Optional[str] = None) -> None:
+    def _configure_model_summary_callback(
+        self, enable_model_summary: bool, weights_summary: Optional[str] = None
+    ) -> None:
+        if not enable_model_summary:
+            return
         if any(isinstance(cb, ModelSummary) for cb in self.trainer.callbacks):
             return
-        if weights_summary is not None:
+        if weights_summary is None:
+            rank_zero_deprecation(
+                "Setting `Trainer(weights_summary=None)` is deprecated in v1.5 and will be removed"
+                " in v1.7. Please set `Trainer(enable_model_summary=False) instead."
+            )
+            return
```

Comment:
isn't the warning below enough? you can do a check below if it's None and return from there itself.

Reply:
I wanted to use a different warning message as `weights_summary=None` -> `enable_model_summary=False` whereas `weights_summary is not None` -> `callbacks=[ModelSummary(...)]`